plugins/services/src/js/components/modals/ServiceDestroyModal.js

```diff
@@ -140,43 +131,149 @@ class ServiceDestroyModal extends React.Component {
+          })}
+          <a
+            href="https://docs.mesosphere.com/service-docs/hdfs/uninstall/"
+            target="_blank"
+            title={intl.formatMessage({
+              id: "COMMON.DOCUMENTATION_TITLE"
+            })}
+          >
+            {intl.formatMessage({
+              id: "COMMON.DOCUMENTATION"
+            })}
+          </a>
+          {intl.formatMessage({
+            id: "SERVICE_ACTIONS.DELETE_SERVICE_FRAMEWORK_2"
+          })}
+        </p>
+        <div className="flush-top snippet-wrapper">
+          <ClickToSelect>
+            <pre className="prettyprint flush-bottom">
+              dcos packages uninstall --app-id={serviceName}
+            </pre>
+          </ClickToSelect>
+        </div>
+        {this.getErrorMessage()}
+      </Modal>
+    );
+  }
 
-    let itemText = "Service";
+  getDestroyServiceModal() {
+    const { onClose, open, service, intl } = this.props;
+    const serviceName = service ? service.getId() : "";
+    let isButtonDisabled = this.state.isButtonDisabled;
+
+    let itemText = `${StringUtil.capitalize(UserActions.DELETE)}`;
 
     if (service instanceof Pod) {
-      itemText = "Pod";
+      itemText += " Pod";
     }
 
     if (service instanceof ServiceTree) {
-      itemText = "Group";
+      itemText += " Group";
     }
 
-    const heading = (
-      <ModalHeading className="text-danger">
-        {StringUtil.capitalize(UserActions.DELETE)} {itemText}
-      </ModalHeading>
-    );
+    if (service instanceof Framework) {
+      isButtonDisabled = false;
+      itemText = "Dismiss";
+    }
 
     return (
       <Confirm
-        disabled={isPending}
-        header={heading}
+        disabled={isButtonDisabled}
+        header={this.getModalHeading()}
         open={open}
         onClose={onClose}
         leftButtonText="Cancel"
         leftButtonCallback={onClose}
-        rightButtonText={`${StringUtil.capitalize(UserActions.DELETE)} ${itemText}`}
+        rightButtonText={itemText}
         rightButtonClassName="button button-danger"
         rightButtonCallback={this.handleRightButtonClick}
         showHeader={true}
       >
-        {this.getDestroyMessage()}
+        <p>
+          {intl.formatMessage({
+            id: "SERVICE_ACTIONS.DELETE_SERVICE"
+          })}
+          <strong>{serviceName}</strong>
+          {intl.formatMessage({
+            id: "SERVICE_ACTIONS.DELETE_SERVICE_2"
+          })}
+        </p>
+        <input
+          className="form-control filter-input-text"
+          onChange={this.handleChangeInputFieldDestroy}
+          type="text"
+          value={this.state.inputFieldDestroy}
+        />
         {this.getErrorMessage()}
       </Confirm>
     );
   }
+
+  getModalHeading() {
+    return (
+      <ModalHeading className="text-danger">
+        {StringUtil.capitalize(UserActions.DELETE)}
+      </ModalHeading>
+    );
+  }
+
+  getSubHeader() {
+    if (!this.props.subHeaderContent) {
+      return false;
```

Comment:
@weblancaster - let's return `null` here to be consistent. We generally use `null` or `<noscript/>` for empty markup :)